packages/components/atoms/f-image-tile/src/components/ImageTile.vue

```diff
@@ -12,20 +47,108 @@ export default {
     name: 'ImageTile',
     components: {},
     props: {
+        href: {
+            type: String,
+            default: null
+        },
+        tileId: {
+            type: String,
+            default: null
+        },
+        isSelected: {
+            type: Boolean,
+            default: false
+        },
+        isLink: {
+            type: Boolean,
+            default: false
+        },
+        displayText: {
+            type: String,
+            default: null
+        },
+        imgSrc: {
+            type: String,
+            default: null
+        }
+    },
+    data () {
+        return {
+            isFilterSelected: false
+        };
+    },
+    watch: {
+        isSelected (newValue) {
+            if (newValue !== this.isFilterSelected) {
+                this.isFilterSelected = newValue;
+            }
+        }
+    },
+    mounted () {
+        this.isFilterSelected = this.isSelected;
+    },
+    methods: {
+        /**
+         * Toggles the isFilterSelected data property
+         * when the checkbox is toggled.
+         *
+         */
+        toggleFilter () {
+            this.isFilterSelected = !this.isFilterSelected;
+
+            this.$nextTick(() => {
+                this.$emit('toggle', this.filterEmitObject(this.tileId));
+            });
```

Comment:
this is nice - hopefully it circumvents any blocking side effects of the event being emitted! 🤞